app/src/server/api/routers/usage.router.ts

```diff
@@ -73,42 +78,39 @@ export const usageRouter = createTRPCRouter({
             fn.sum(sql<number>`case when ul.billable = false then 0 else ul."cost" end`).as("cost"),
             fn.count("ul.id").as("numQueries"),
           ])
+          .where(sql`"ul"."createdAt"`, ">=", input.startDate)
+          .where(sql`"ul"."createdAt"`, "<=", input.endDate)
           .executeTakeFirst(),
         finetunesQuery.select("ft.createdAt").execute(),
       ]);
 
-      let originalDataIndex = periods.length - 1;
-      // *SLAMS DOWN GLASS OF WHISKEY* timezones, amirite?
-      let dayToMatch = dayjs(input.startDate || new Date());
-      // Ensure that the initial date we're matching against is never before the first period
-      if (
-        periods[originalDataIndex] &&
-        dayToMatch.isBefore(periods[originalDataIndex]?.period, "day")
+      // Fillin in missing periods
+      const startDate = dayjs(input.startDate);
+      const endDate = dayjs(input.endDate);
+
+      const allDates: Date[] = [];
+      for (
+        let date = startDate;
+        date.isBefore(endDate) || date.isSame(endDate);
+        date = date.add(1, "day")
       ) {
-        dayToMatch = dayjs(periods[originalDataIndex]?.period);
+        allDates.push(date.toDate());
```

Comment:
Curious what happens if we skip this? I'd have assumed that even if the data is sparse (ie we only include entries for the days when there was actual usage) that the graphing library could handle that properly by just not showing any output on that day. But totally could be wrong about that!

Reply:
The library cuts the graph in this case, displaying only a period that has some data.
It will work only if we have some actual data on day 1 and day 31.
It would show an empty graph without any dates for new projects without any usage.

Reply:
Ah yes I see why you did that. Although I think the code could be simplified and made a bit faster by using the `domain` argument of the XAxis component and just setting the min/max dates there. I can't find good documentation on the recharts page, but it does look like it's part of the API.

https://recharts.org/en-US/api/XAxis#domain
https://github.com/recharts/recharts/issues/1809 (workarounds for dates listed there)